### DISCUSSION TITLE
Is there a way to use environment variables inside .env to expand to its value ?

### DISCUSSION THREAD

**Discussion by sharadshahihw** [runtime-env/import-meta-env#946] [Q&A]
We have use case to use environment variables that are created by kubernetes container.
Is there a way to use process.env variables to expand inside .env file?

```.env
REACT_APP_HELLO=$USERNAME
```

I am looking similar to what react-env does
`npx react-env --path .env --dest dist`

The output will replace $USERNAME with environment variable value.

**Comment by soc221b** [comment]
You could use [@import-meta-env/prepare](https://iendeavor.github.io/import-meta-env/guide/tools/prepare.html) (with `-u, --user-environment`) to dynamically generate `.env` from something like `.env.defaults` and `process.env`. 😄 

See also: https://iendeavor.github.io/import-meta-env/api.html#import-meta-env-prepare .
